backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java

```diff
@@ -159,10 +159,20 @@ private Channel tryBorrowFromPool() {
         return Mono.fromCallable(this::borrowFromPool)
             .doOnError(throwable -> LOGGER.warn("Cannot borrow channel", throwable))
             .retryBackoff(MAX_BORROW_RETRIES, MIN_BORROW_DELAY, FOREVER, Schedulers.elastic())
+            .onErrorMap(this::propagateException)
```

Comment:
I need a reactor course here :)
you have `mono.doOnError.retry.onErrorMap`.
Does it mean that:
- doOnError will be called each time the initial borrowFromPool() function throws? Thus display a warning
- onErrorMap will throw only when all retries are exhausted?
If it is that I approve this change :)

Reply:
exactly!